Makefile.common

```diff
@@ -243,14 +238,9 @@ TAC = $(shell which tac >/dev/null 2>&1 && echo "tac" || echo "tail -r")
 
 ALL_CMX_FILES = $(subst obj/Lib_Buffer.cmx,obj/Lib_Memzero0.cmx obj/Lib_Buffer.cmx,$(patsubst %.ml,%.cmx,$(shell echo $(ALL_ML_FILES) | $(TAC))))
 
-ifeq ($(OS),Windows_NT)
-  export OCAMLPATH := $(FSTAR_HOME)/lib;$(OCAMLPATH)
-else
-  export OCAMLPATH := $(FSTAR_HOME)/lib:$(OCAMLPATH)
-endif
-
 # Warning 8: this pattern-matching is not exhaustive.
 # Warning 20: this argument will not be used by the function.
 # Warning 26: unused variable
-OCAMLOPT = OCAMLFIND_IGNORE_DUPS_IN="`ocamlc -where`/compiler-libs" ocamlfind opt -package fstar.lib -linkpkg -thread -g -I $(HACL_HOME)/obj -w -8-20-26
-OCAMLSHARED = OCAMLFIND_IGNORE_DUPS_IN="`ocamlc -where`/compiler-libs" ocamlfind opt -shared -package fstar.lib -thread -g -I $(HACL_HOME)/obj -w -8-20-26
+export OCAMLFIND_IGNORE_DUPS_IN="`ocamlc -where`/compiler-libs"
+OCAMLOPT = $(FSTAR_EXE) --ocamlenv ocamlfind opt -package fstar.lib -linkpkg -thread -g -I $(HACL_HOME)/obj -w -8-20-26
```

Comment:
I thought we had had a discussion before on the topic of running  the ocaml tools via F*, and we didn't feel so enthusiastic on the basis that it would be yet one more level of indirection to debug. Maybe @R1kM remembers? Can you say whether there's any intended benefit to doing this?

Reply:
I think the relevant discussion was here: https://github.com/hacl-star/hacl-star/pull/998

Reply:
Right, I understood that discussion as "let's not have F* drive the compilation" (`fstar.exe --ocamlopt`) but I was thinking that `fstar.exe --ocamlenv` is fine, it's just asking F* to extend the environment with a relevant `OCAMLPATH`. I'm fine removing this chunk and just extending OCAMLPATH with `fstar.exe --locate_ocaml` though (instead of $FSTAR_HOME/lib, there's no FSTAR_HOME).

Reply:
If you want F* to configure the OCaml environment (which I think is reasonable), a good way might be like other tools do, to have `fstar --env` print something like:

```
export OCAMLPATH=...
export OCAML_IGNORE_DUPS_IN=...
etc.
```

to be `eval`'d in build scripts and in the user's environment. What do you think?

Reply:
that way this is trivially debuggable rather than having to go through the F* source code the minute something gets wrong

Reply:
excellent! if you want to use that, then, I'm fine with it -- this is immediately debuggable and we don't have to rely on F* to fork processes (which introduces additional opportunities for unexpected behavior)